### PR TITLE
Delete batch publications

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
1. New feature to allow users to delete more than 1 publication at once.  
2. After import publications, revert the header color of all the imported publications to the original color at the same time. 

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 
- [x] Improvemnt  

# Problem
1. User needs to delete publication once at a time
2. After import publications, only the publications on the current page revert back to the original color after 2.5 seconds, the  publications that aren't on the current page still have the props of 'newlyAdded', which cause the header to remain green when the user clicks on that page  

# Solution description
1. Add deleteBulkPublications endpoint to handle deleting more than one publication at a time. 
2. Dispatching an action which removes the 'newlyAdded' props for all publications after 2.5 seconds 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 

# Risk
Low

### CODE REVIEW

**Comment by vgoh1** [comment]
Does this happen on your end? After creating a new publication, the new publication doesn't show up in the publications list anymore despite the number of publications changing. It shows up after I refresh the page though and after I import publications.

**Review by cyr17 — APPROVED** [review]
looks good!

**Review by vgoh1 — APPROVED** [review]
The issue I was having is gone, so LGTM.

**Review by rpra0006 — CHANGES_REQUESTED** [review]
Delete publications button doesn't work when I click on it
image

**Comment by pmyip3** [comment]
> Delete 10 publications button doesn't work when I click on it. Single publication delete works though.
> image

No idea why the onClick function was removed, but it should be fixed now, thanks for testing it.

**Review by rpra0006 — APPROVED** [review]
All working now

File: api/src/controllers/publications.js [hunk 1]
```diff
@@ -314,11 +314,30 @@ function importPublications(req, res, next) {
...
 
+/**
+ * Handles a PATCH request to delete a list of publication by the mongo object id on the endpoint /publications/
+ *
+ * @param req request object - the list of publication ids given in the body
+ * @param res response object
+ * @returns 200: publications deleted successfully
+ * @returns 400: error deleting publications
+ */
+async function deleteBulkPublications(req, res, next) {
```
Comment (aksh0001) [inline api/src/controllers/publications.js:325]: nit with function name. I think `batch` delete is more common than `bulk` delete in terms of naming this kind of functionality (as with the PR name too). Also, curious about why `PATCH` and why not `DELETE`?

File: client/src/actions/publications.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import {
   IMPORT_FAIL,
   IMPORT_END,
   IMPORT_EMPTY,
+  DELETE_BULK_PUBLICATIONS,
```
Comment (aksh0001) [inline client/src/actions/publications.js:13]: Same naming here if we go with -> `BATCH`

Comment (aksh0001) [reply client/src/actions/publications.js:13]: ditto everywhere else

File: client/src/actions/publications.js [hunk 3]
```diff
@@ -151,3 +153,16 @@ export const createBulkPublications = (teamId, publicationList) => async (dispat
...
 };
+
+export const deleteBulkPublications = (publicationIdList) => async (dispatch) => {
+  try {
+    await api.deleteBulkPublications(publicationIdList);
+    dispatch({
+      type: DELETE_BULK_PUBLICATIONS,
+      payload: publicationIdList,
+    });
+    dispatch(successMessageCreator(`${publicationIdList.length} publication(s) has been deleted`));
```
Comment (aksh0001) [inline client/src/actions/publications.js:164]: nit `has` -> `have`

File: client/src/components/publications/Publications.js [hunk 4]
```diff
@@ -98,6 +99,15 @@ const Publications = () => {
   useEffect(() => {
     const sortedPublication = sortPublications(teamPublications, options.sortBy);
     setPublications(sortedPublication);
+    const newlyAddedPublications = teamPublications.filter((pub) => pub.newlyAdded === true);
```
Comment (aksh0001) [inline client/src/components/publications/Publications.js:102]: do we need to `=== true`? or can we omit it and just have `(pub) => pub.newlyAdded`? Also, this is a separate issue, but newlyAdded could be renamed to `isNewlyAdded` to make it more clear what we're referring to.

**Review by aksh0001 — APPROVED** [review]
LGTM, only minors/nits

File: api/src/controllers/publications.js (hunk 1 above)
Comment (pmyip3) [reply api/src/controllers/publications.js:325]: cause the delete api call only accept a parameter (the deleted document id), but do not accept body

**Pull request merged**